### PR TITLE
Implemented email and sms rendering services

### LINKED ISSUE #200: Create SMS and Email rendering services
### Description
Create SMS and Email rendering services with necessary infrastructure

### Requirements
- project should be runnable
- rendering must be successful should show validation errors

### Deliverables
- `SmsRenderingService` and `EmailRenderingService`
- validation for messages to render
- register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by Habiba67y**
Closes #200

### CODE REVIEW

**Review by abdura52 — DISMISSED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Notifications/Models/EmailMessage.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/// <summary>
+/// Represents an email message for notifications
+/// </summary>
+public class EmailMessage : NotificationMessage
+{
+    /// <summary>
+    /// Gets or sets email address of sender user
+    /// </summary>
+    public string SendEmailAddress { get; set; } = default!;
```
[the remark this answers is not in the document]
Comment (sulton-max) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/Notifications/Models/EmailMessage.cs:13]: Menda ketgan xato birga bir takrorlanibdi ))

**Review by Ilxom-22 — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/EmailRenderingService.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using AirBnB.Application.Common.Settings;
+using AirBnB.Domain.Enums;
+using FluentValidation;
+using Microsoft.Extensions.Options;
+using System.Text.RegularExpressions;
+using System.Text;
+
+namespace AirBnB.Infrastructure.Common.Notifications.Services;
+
+public class EmailRenderingService(
```
[the remark this answers is not in the document]
Comment (Habiba67y) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/EmailRenderingService.cs:12]: to the interface?

[the remark this answers is not in the document]
Comment (sulton-max) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/EmailRenderingService.cs:12]: Nope, to the implementation

[the remark this answers is not in the document]
Comment (Habiba67y) [reply src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Notifications/Services/EmailRenderingService.cs:12]: oh sorry, I'll fix it

**Review by sulton-max — APPROVED** [review]

**Review by Ilxom-22 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request merged**